src/briefcase/platforms/iOS/xcode.py

```diff
@@ -338,6 +341,21 @@ def run_app(self, app: BaseConfig, udid=None, **kwargs):
         except subprocess.CalledProcessError:
             raise BriefcaseCommandError(f"Unable to install new version of app {app.app_name}.")
 
+        # Start log stream for the app.
+        simulator_log_popen = self.subprocess.Popen(
+            [
+                "xcrun", "simctl", "spawn", udid,
+                "log", "stream",
+                "--style", "compact",
+                "--predicate", f'senderImagePath ENDSWITH "/{app.formal_name}"'
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            bufsize=1,
+        )
+        time.sleep(3)  # let log stream start up
```

Comment:
I'm not sure how long to wait here....ideally, i suppose, it'd be nice to know when `log stream` is ready....but I don't think that's an option...

Reply:
So, on my exceptionally slow virtualized instance of macOS, I did encounter an instance of missing the beginning of the log when this wait was 1 second.

While starting `log stream` before running the app at least makes it possible to display the entire log, I believe there is still a race condition. When `Popen` is invoked, it simply forks a process with instructions to run a command; so, I'm not sure there would necessarily be any guarantee of the state of that command once `Popen` returns control.

So, if `log stream` (for some bizarre reason) takes 10 seconds to start actually capturing the log entries, it's quite likely that the app will already be up and running. That's why, for instance, it would be nice if we could enter another loop sending `SIGUSR1` signals or something to `log stream` until it responded affirmatively that it's ready to actually stream logs. Then we'd know 100% it is safe to start the app and show all the log entries.

side note: i found this in `Popen` ;) thought it somewhat apt
```python
        # How long to resume waiting on a child after the first ^C.
        # There is no right value for this.  The purpose is to be polite
        # yet remain good for interactive users trying to exit a tool.
        self._sigint_wait_secs = 0.25  # 1/xkcd221.getRandomNumber()
```

Reply:
Ah - that makes sense. I guess we need to make the call between slow enough to guarantee that *really* slow machines will have started the log, and being slow enough to get in the way of app development. 

My gut tells me that 3 seconds is leaning a little to hard on the "accomodate slow machines" side; made worse by the fact that it's a 3 second wait before any other console output happens (so it's *right* on the cusp of a "did that work?" wait). If you're running macOS virtualised, you're definitely going to be seeing the worst case scenario, and most people doing macOS development won't be doing so in a virtualised container. I wonder if the 0.25 value you've found actually might be good enough in this case...

Reply:
I found that `0.25` seconds did work sometimes on my virtualized macOS...maybe 50/50. As you said though, this should definitely be the extreme example of delay on macOS....I can only even allocate a single CPU to the VM without a kernel panic :| so I'm fine with `0.25`....can always increase later.